vmware/vmware-cloud-director-ui-components
# Data Exporter initial implementation

## Description
A dialog to export data

* Allows the UI to select columns to be exported
* Provides a progress indicator
* Converts the data that is fetched by the caller into a CSV

## Testing Done

* Go to main application -> Click Data Exporter Button
* Make sure Select All is disabled when all columns are selected
* Make sure Select All selects all columns when clicked
* Export is disabled when no columns are selected
* Exporting downloads a file, and headers are the field display names
* Dialog disappears after download finishes
* Dialog disappears when user clicks export

## Not yet done

* Unit tests
* CSV injection prevention

Signed-off-by: Juan Mendes <jmendes@vmware.com>
